packages/@lwc/synthetic-shadow/src/faux-shadow/shadow-root.ts

```diff
@@ -117,6 +117,13 @@ export function isHostElement(elm: Element | Node): boolean {
 
 let uid = 0;
 
+type ShadowRootMode = 'closed' | 'open';
+
+interface ShadowRootInit {
```

Comment:
typescript has this definition already! why do we need to redefine that?

Reply:
exactly what's tripping me up. if we don't redefine in this file then the `ShadowRootMode` comes in as type `any` on the param (can see in the photos), but if we move it here locally then it's type checked as expected. could this be a symptom of a bigger issue?

Reply:
Agreed with @caridy, this looks strange. @sfdc-jkovitz Do you mind sharing a standalone repo to double-check?

Reply:
@pmdartus @caridy ok the screenshotted issue with the types coming thru as `any` is probably related to how TS is setup. If I goto https://github.com/salesforce/lwc/blob/master/packages/@lwc/engine-dom/src/renderer.ts#L112 and alter/omit the `ShadowRootInit` then I get type errors as expected for `attachShadow`.

Reply:
@caridy I spoke w/ PM re: this and we found that Ravi had made some updates since the issue was created:
https://github.com/salesforce/lwc/blame/master/packages/%40lwc/synthetic-shadow/src/faux-shadow/element.ts#L79
It looks like he added a check to determine if we should use our own attachShadow function, or the native one, and since our attachShadow function is never called by end-users the typing scenario should be ok as established. Are you alright with cancelling out this PR and closing the work item?

Reply:
yeah!